### PR TITLE
Backup our new thewca github organization.

### PR DESCRIPTION
**Pull request by jfly**
@lgarron, could you take a look at this? I've already deployed it to staging.worldcubeassociation.org, so you can check it out in action there as well.

### CODE REVIEW

File: chef/site-cookbooks/wca/recipes/cronjobs.rb [hunk 1]
```diff
@@ -1,24 +1,29 @@
...
 
 admin_email = "admin@worldcubeassociation.org"
 path = "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games"
 
-db_dump_folder = "#{repo_root}/secrets/wca_db"
-dump_command = "#{repo_root}/scripts/db.sh dump #{db_dump_folder}"
+secrets_folder = "#{repo_root}/secrets"
+db_dump_folder = "#{secrets_folder}/wca_db"
+dump_db_command = "#{repo_root}/scripts/db.sh dump #{db_dump_folder}"
+dump_gh_command = "github-backup --incremental --all -t #{secrets['GITHUB_BACKUP_ACCESS_TOKEN']} --organization thewca -o #{secrets_folder}/github-thewca"
```
Comment (lgarron) [inline chef/site-cookbooks/wca/recipes/cronjobs.rb]: Is it possible we might have any forks in the future? I'd suggest passing `--fork` just in case.

(Same for `--private`, except that `thewca` doesn't have private repos, I assume.)

I also notice you added `--incremental`, which I forgot to include in my email. Good on you. :-)

File: chef/data_bags/secrets/production.json [hunk 2]
```diff
@@ -1,68 +1,74 @@
 {
   "id": "production",
   "_password_comment": {
-    "encrypted_data": "qHlXr8rirW3ySV9z+pP7rYeGbnxCAcgLtv5vYZ6U/EeCMUsSQDeqM9lq0pHW\nCC+ZGlxJr9XY1EGZtKFUOLl4D7eUoC7c3rJu2hWfx8Pr4gEQB3AVsxa8XUaM\nIs+/C5VPB8LjIoTOBusYswrEMmVSRA==\n",
-    "iv": "ogGQgMdkrA7EzQXx0b6SEw==\n",
+    "encrypted_data": "/ce5YkvxK69ni8NFJWErjskGJfwmVwUyKu/yxEvrlq33CHZa3VeuNDgIUzFY\nd35YL2asqpFFOCBkSbVbEIcycD7zsuSTQkSFSiEmPEtTbLUbrNVKGPYD4CLF\nMvz45X+zwHjr7BAoSJ1BWZgx0Mof8A==\n",
```
Comment (lgarron) [inline chef/data_bags/secrets/production.json]: If these constant changes are unrelated to `GITHUB_BACKUP_ACCESS_TOKEN`, I'd try to put them in a separate commit.

File: chef/site-cookbooks/wca/recipes/cronjobs.rb [hunk 3]
```diff
@@ -1,24 +1,29 @@
...
+backup_command = "#{dump_db_command} && #{dump_gh_command}"
```
Comment (lgarron) [inline chef/site-cookbooks/wca/recipes/cronjobs.rb]: I would probably call these `backup_db_command` and `backup_gh_command` to avoid different terminology, but I don't see a problem with `dump.

Comment (jfly) [inline chef/site-cookbooks/wca/recipes/cronjobs.rb]: The different terminology is actually intentional: I'm making a distinction between "dumping to the filesystem" and "backing up to tarnsap" here.

File: chef/site-cookbooks/wca/recipes/cronjobs.rb (hunk 1 above)
Comment (jfly) [inline chef/site-cookbooks/wca/recipes/cronjobs.rb]: Good call! I've added both `--private` and `--fork`.

File: chef/data_bags/secrets/production.json (hunk 2 above)
Comment (jfly) [inline chef/data_bags/secrets/production.json]: They actually didn't change, it's just a side effect of the encryption... when you change one, they all change.

**Comment by jfly** [comment]
I've merged this up and deployed it. Some notes about deplying it here:

```
~ @production> crontab -l
# Chef Name: db backup
MAILTO=admin@worldcubeassociation.org
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
0 0 * * 1 /home/cubing/worldcubeassociation.org/scripts/db.sh dump /home/cubing/worldcubeassociation.org/secrets/wca_db && /home/cubing/worldcubeassociation.org/scripts/backup.sh
# Chef Name: hourly schedule rails work
MAILTO=admin@worldcubeassociation.org
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
0 * * * * (cd /home/cubing/worldcubeassociation.org/WcaOnRails; RACK_ENV=production bin/rake work:schedule)
# Chef Name: cronned results scripts
MAILTO=admin@worldcubeassociation.org
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
CONTENT_TYPE=text/html
CONTENT_TRANSFER_ENCODING=utf8
0 4 * * 1,3,5 /home/cubing/worldcubeassociation.org/scripts/cronned_results_scripts.sh
```

Running chef:
```
~ @production> sudo chef-solo -o 'role[wca]' -l debug -L /tmp/chef-debug.log
...
Running handlers:
Running handlers complete
Chef Client finished, 13/147 resources updated in 39.50865415 seconds
```

After running chef, you may notice that we still have the old "db backup" cron job, in addition to the new "backup" cron job. I've removed the old one manually (chef isn't very good about cleaning up after old runs, we really need to be careful when running it on an existing server... the safest thing to do would be to always spin up a new server instead of trying to run chef on an existing server).

```
~ @production> crontab -l
# Chef Name: db backup
MAILTO=admin@worldcubeassociation.org
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
0 0 * * 1 /home/cubing/worldcubeassociation.org/scripts/db.sh dump /home/cubing/worldcubeassociation.org/secrets/wca_db && /home/cubing/worldcubeassociation.org/scripts/backup.sh
# Chef Name: hourly schedule rails work
MAILTO=admin@worldcubeassociation.org
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
0 * * * * (cd /home/cubing/worldcubeassociation.org/WcaOnRails; RACK_ENV=production bin/rake work:schedule)
# Chef Name: cronned results scripts
MAILTO=admin@worldcubeassociation.org
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
CONTENT_TYPE=text/html
CONTENT_TRANSFER_ENCODING=utf8
0 4 * * 1,3,5 /home/cubing/worldcubeassociation.org/scripts/cronned_results_scripts.sh
# Chef Name: backup
MAILTO=admin@worldcubeassociation.org
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
0 0 * * 1 /home/cubing/worldcubeassociation.org/scripts/db.sh dump /home/cubing/worldcubeassociation.org/secrets/wca_db && github-backup --incremental --fork --private --all -t bf88e06ea2985e31d41cd045f82d3256911fc66a --organization thewca -o /home/cubing/worldcubeassociation.org/secrets/github-thewca&& /home/cubing/worldcubeassociation.org/scripts/backup.sh
```

File: chef/data_bags/secrets/production.json (hunk 2 above)
Comment (lgarron) [inline chef/data_bags/secrets/production.json]: Cool, good to know.

File: chef/site-cookbooks/wca/recipes/cronjobs.rb (hunk 1 above)
Comment (lgarron) [inline chef/site-cookbooks/wca/recipes/cronjobs.rb]: Does the token have access to private repos?

Comment (jfly) [inline chef/site-cookbooks/wca/recipes/cronjobs.rb]: Nope just `public_repo`: image
